wgsl/index.bs

```diff
@@ -689,84 +682,240 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
 The terms defined in this section express counts of 8-bit bytes.
 
 We will use the following notation:
+* <dfn noexport>AlignOf</dfn>(|T|) is the alignment of host-shareable type |T|.
+* AlignOf(|S|, |M|) is the alignment of member |M| of the host-shareable structure |S|.
+* <dfn noexport>SizeOf</dfn>(|T|) is the size of host-shareable type |T|.
+* SizeOf(|S|, |M|) is the size of member |M| of the host-shareable structure |S|.
+* <dfn noexport>StrideOf</dfn>(|A|) is the element stride of host-shareable array type |A|.
+* <dfn noexport>OffsetOf</dfn>(|S|, |M|) is the offset of member |M| from the start of the host-shareable structure |S|.
+
+
+#### Type Alignment and Size ####  {#type-alignment-and-size}
+
+Every [=host-shareable=] data type has an alignment and size value.
```

Comment:
Maybe `type has  a default alignment and size` ?

Reply:
I'm hesitant to use `default` here. It suggests the **type** can have a different size / alignment, which we have decided to drop for the initial version in https://github.com/gpuweb/gpuweb/issues/1393

Reply:
Maybe something like "Every [=host-shareable=] data type has a  default alignment and size value. The alignment and size values for a given strut member can differ from the defaults if the `align` and/or `size` decorations are used." ?

Or maybe this isn't the right place for that distinction?

Reply:
Done